f8a_worker/workers/recommender.py

```diff
@@ -514,9 +514,13 @@ class RecommendationV2Task(BaseTask):
 
     def call_pgm(self, payload):
         """Calls the PGM model with the normalized manifest information to get the relevant packages"""
-        pgm_url = PGM_URL_REST + "/api/v1/schemas/kronos_scoring"
         try:
-            if payload is not None:
+            # TODO remove hardcodedness for payloads with multiple ecosystems
+            if payload is not None and 'ecosystem' in payload[0]:
```

Comment:
You could say `if payload and 'ecosystem' in payload[0]:` - that way you would catch scenario when `payload` variable is an empty array.